dev/Tiles/metatile.h

```diff
@@ -0,0 +1,60 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include "../../lib/SMSlib.h"
+#include "tile.h"
+
+typedef struct MetaTile
+{
+    struct Tile tiles[16];
+    unsigned char numTiles;
+    unsigned char numTilesX;
+    unsigned char numTilesY;
+} MetaTile;
+
+MetaTile* MetaTile_Create
+(
+    Tile tiles[16],
+    unsigned char numTiles,
+    unsigned char numTilesX,
+    unsigned char numTilesY
+) {
+    unsigned char tileXY = numTilesX * numTilesY;
+
+    if(tileXY > 16)
+    {
+        // We have an error here
+    }
+    unsigned char numMetatiles = numTiles / tileXY;
+    
+    MetaTile* metatile = malloc(sizeof (struct MetaTile));
+    metatile->numTilesX = numTilesX;
+    metatile->numTilesY = numTilesY;
+    metatile->numTiles = numTiles;
+
+    unsigned char i;
+    for(i = 0; i < metatile->numTiles; ++i)
+    {
+        metatile->tiles[i] = tiles[i];
```

Comment:
Did you realise this is copying the data of each tile rather than simply copying a reference to the tile?
I think that's probably okay for you, but since you malloc-ed your Tile in the first place it does seem kind of funny to have it take up two spots in memory!

Reply:
EDIT: looking at your `main.c` file it's a good job you are doing a value copy here, because your tile data is local to the `loadGraphics2vram` function, so copying a reference would experience problems due to dangling pointers.

That confused me though, and I wondered when you ever call `CreateTile`, but looks like you don't, maybe for this reason!

It's tough keeping track of when you want to be copying memory or taking pointers and whether that's intentional or not. APIs in C usually have to be quite strict about maintaining conventions so that consumers understand the implications of what they are passing.

To this point, I would certainly recommend getting your head around the `const` keyword in C, which does help prevent a fair amount of needless confusion / bugs as well as quite often making compilation more efficient.